project/Build.scala

```diff
@@ -1615,6 +1626,63 @@ object Build {
       scalaJSLinkerConfig ~= { _.withSemantics(TestSuiteLinkerOptions.semantics _) },
       scalaJSModuleInitializers in Test ++= TestSuiteLinkerOptions.moduleInitializers,
 
+      /* The script that calls setExportsNamespaceForExportsTest to provide
+       * ExportsTest with a loopback reference to its own exports namespace.
+       * Only when using an ES module.
+       * See the comment in ExportsTest for more details.
+       */
+      setModuleLoopbackScript in Test := Def.settingDyn[Task[Option[FileVirtualBinaryFile]]] {
```

Comment:
I really don't like this, but I don't have a better idea either :(

Reply:
Hah, turns out this works:

`test.mjs`
```javascript
export const blah = 1;

import loopback from './loopback.mjs';

console.log(loopback.blah);
```

`loopback.mjs`
```javascript
import * as test from './test.mjs';

export default test;
```

Did you try this? (not sure it works in the more involved setting we have).

Reply:
That looks more elegant, indeed.

I'm not sure how to set it up, though. The biggest issue is that we have fastOpt and fullOpt. The loopback module must be different for each, since it must import a different `testsuite-*opt.mjs`. And then that means that the `@JSImport("loopback.mjs", ...)` inside the test suite must be different for fastOpt and fullOpt. And that is quite annoying to set up. We also need to avoid the mere presence of that `@JSImport` when using NoModule or CommonJS. We've precedent for something similar, but it's yet another burden.

I'm also not *convinced* that the circular import trick does not suffer from the asynchronous theoretical problem. After all, circular bindings need to be initialized in *some* order. I would have to dig quite deep into the spec to ascertain that.

I'm not sure it's worth the trouble.

Reply:
Aight, fair enough. Let's go ahead with this and revisit if we encounter issues.